src/main/app/forms/validation/validators/extraFormFieldsArePopulated.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  registerDecorator,
+  ValidationArguments,
+  ValidationOptions,
+  ValidatorConstraint,
+  ValidatorConstraintInterface
+} from 'class-validator'
+
+@ValidatorConstraint()
+export class ExtraFormFieldsArePopulatedConstraint implements ValidatorConstraintInterface {
+  validate (value: any, args?: ValidationArguments) {
+    return args.constraints.every(field => !!args.object[field])
+  }
+}
+
+/**
+ * Verifies if all of the fields in given object are populated with "truthy" value.
+ */
+export function ExtraFormFieldsArePopulated (fields: string[], validationOptions?: ValidationOptions) {
```

Comment:
What about `FieldsArePopulated`? I personally do not understand this `extra form` bit. Why extra? Form bit is also unnecessary as validation should usable anywhere, not just for the form data.